--- a/frontvm3-20060623/Makefile-C	2014-11-14 07:37:29.430146785 +0100
+++ b/frontvm3-20060623/Makefile-C	2014-11-14 07:37:54.728146745 +0100
@@ -1,5 +1,5 @@
-export CFLAGS = -O2 -g -Wall -DOGG_MUSIC
-export LIBS = -lGL -lGLU -lvorbisfile -lvorbis -logg
+export CFLAGS = -march=native -pipe -O2 -Wall -DOGG_MUSIC
+export LIBS = -lGL -lGLU -lvorbisfile -lvorbis -logg -lm
 export FE2OBJS = ../fe2.part1.o ../fe2.part2.o
 
 THIS=Makefile-C
@@ -24,11 +24,8 @@
 
 fe2obj:
 	as68k/as68k --output-c fe2.s
-	# this bit can be optimised because it is lots of small functions
-	$(CC) -DPART1 -O1 -fomit-frame-pointer -Wall -Wno-unused -s `sdl-config --cflags` -c fe2.s.c -o fe2.part1.o
-	# this can't unless you have shitloads of memory and a fuck meaty
-	# machine, because it is a huge stinking function.
-	$(CC) -DPART2 -O0 -fomit-frame-pointer -Wall -Wno-unused -s `sdl-config --cflags` -c fe2.s.c -o fe2.part2.o
+	$(CC) -DPART1 $(CFLAGS) -fomit-frame-pointer -Wall -Wno-unused -s `sdl-config --cflags` -c fe2.s.c -o fe2.part1.o
+	$(CC) -DPART2 $(CFLAGS) -fomit-frame-pointer -Wall -Wno-unused -s `sdl-config --cflags` -c fe2.s.c -o fe2.part2.o
 
 clean:
 	$(MAKE) -C src/ clean
